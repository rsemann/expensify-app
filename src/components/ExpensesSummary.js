import React from 'react';
import {connect} from 'react-redux';
import getVisibleExpenses from '../selectors/expenses';
import getExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = (props) => {
    return (
        <div>
            <p>{`Viewing ${props.count} expenses totalling ${numeral(props.total/1000).format("$0,0.00")}`}</p>
        </div>
    );
};

const mapStateToProps = (state) => {
    const expenses = getVisibleExpenses(state.expenses, state.filters);
    return {
        count: expenses.length,
        total: getExpensesTotal(expenses)
    }
};

export default connect(mapStateToProps)(ExpensesSummary);