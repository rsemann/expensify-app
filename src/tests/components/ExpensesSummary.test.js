import React from 'react';
import {ExpensesSummary} from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';
import {shallow} from 'enzyme';

test('Should render view one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[expenses[0]]} total={expenses[0].amount}/>);
    expect(wrapper).toMatchSnapshot();
});

test('Should render view two expenses', () => {
    const wrapepr = shallow(<ExpensesSummary expenses={[expenses[0], expenses[1]]} total={expenses[0].amount + expenses[1].amount}/>);
    expect(wrapepr).toMatchSnapshot();
});