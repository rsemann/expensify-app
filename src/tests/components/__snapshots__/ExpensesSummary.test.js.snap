// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render view one expense 1`] = `
<div>
  <p>
    Viewing 1 expenses totalling $0.20
  </p>
</div>
`;

exports[`Should render view two expenses 1`] = `
<div>
  <p>
    Viewing 2 expenses totalling $109.70
  </p>
</div>
`;
